src/SFA.DAS.Tasks.Worker.UnitTests/MessageProcessors/ApprovedTransferConnectionInvitationMessageProcessorTests/WhenAnApprovedTransferConnectionInvitationMessageISProcessed.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.EmployerAccounts.Events.Messages;
+using SFA.DAS.Messaging.Interfaces;
+using SFA.DAS.NLog.Logger;
+using SFA.DAS.Tasks.Application.Commands.SaveTask;
+using SFA.DAS.Tasks.API.Types.Enums;
+using SFA.DAS.Tasks.Worker.MessageProcessors;
+
+namespace SFA.DAS.Tasks.Worker.UnitTests.MessageProcessors.ApprovedTransferConnectionInvitationMessageProcessorTests
+{
+    public class WhenAnApprovedTransferConnectionInvitationMessageIsProcessed
+    {
+        private Mock<IMessageSubscriberFactory> _subscriptionFactory;
+        private Mock<IMessageSubscriber<ApprovedTransferConnectionInvitationEvent>> _subscriber;
+        private ApprovedTransferConnectionInvitationEvent _messageContent;
+        private Mock<IMessage<ApprovedTransferConnectionInvitationEvent>> _mockMessage;
+        private Mock<IMediator> _mediator;
+        private CancellationTokenSource _tokenSource;
+        private ApprovedTransferConnectionInvitationMessageProcessor _processor;
+
+        [SetUp]
+        public void Arrange()
+        {
+            _subscriptionFactory = new Mock<IMessageSubscriberFactory>();
+            _subscriber = new Mock<IMessageSubscriber<ApprovedTransferConnectionInvitationEvent>>();
+
+            _messageContent = new ApprovedTransferConnectionInvitationEvent();
+         
+            _mockMessage = new Mock<IMessage<ApprovedTransferConnectionInvitationEvent>>();
+            _mockMessage.Setup(x => x.Content).Returns(_messageContent);
+
+            _mediator = new Mock<IMediator>();
+            _tokenSource = new CancellationTokenSource();
+
+            _processor = new ApprovedTransferConnectionInvitationMessageProcessor(_subscriptionFactory.Object, Mock.Of<ILog>(),
+                _mediator.Object);
+
+            _subscriptionFactory.Setup(x => x.GetSubscriber<ApprovedTransferConnectionInvitationEvent>()).Returns(_subscriber.Object);
+
+            _subscriber.Setup(x => x.ReceiveAsAsync())
+                .ReturnsAsync(() => _mockMessage.Object)
+                .Callback(() => { _tokenSource.Cancel(); });
```

Comment:
This can just be a lambda rather than an anonymous method 
i.e.
Callback(() => _tokenSource.Cancel() );

Reply:
Fixed